packages/flutter/lib/src/animation/curves.dart

```diff
@@ -103,6 +109,20 @@ double _bounce(double t) {
   return 7.5625 * t * t + 0.984375;
 }
 
+/// A curve that inverts its given curve.
+class InvertedCurve implements Curve {
+  final Curve curve;
+  InvertedCurve(this.curve);
+  double transform(double t) => curve(1.0 - t);
```

Comment:
This isn't legit either.  If curve maps 0.0 to 0.0 and 1.0 to 1.0, this new curve will violate that invariant.